src/jabs/ui/main_window.py

```diff
@@ -421,6 +429,37 @@ def _project_load_error_callback(self, error: Exception):
         QtWidgets.QMessageBox.critical(
             self, "Error loading project", str(error))
 
+    def _clear_cache_action(self):
+        """
+        Clear the cache for the current project. Opens a dialog to get user confirmation first.
+        """
+
+        app = QtWidgets.QApplication.instance()
+        dont_use_native_dialogs = QtWidgets.QApplication.instance().testAttribute(
+            Qt.ApplicationAttribute.AA_DontUseNativeDialogs)
+
+        # if app is currently set to use native dialogs, we will temporarily set it to use Qt dialogs
+        # the native style, at least on macOS, is not ideal so we'll force the Qt dialog instead
+        if dont_use_native_dialogs is False:
+            app.setAttribute(Qt.ApplicationAttribute.AA_DontUseNativeDialogs, True)
+
+        result = QtWidgets.QMessageBox.warning(
+            self,
+            "Clear Cache",
+            "Are you sure you want to clear the project cache?",
+            QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
+            QtWidgets.QMessageBox.No
+        )
+
+        # restore the original setting
+        app.setAttribute(Qt.ApplicationAttribute.AA_DontUseNativeDialogs, dont_use_native_dialogs)
+
+        if result == QtWidgets.QMessageBox.Yes:
+            self._project.clear_cache()
+            # need to reload the current video to force the pose file to reload
+            self._central_widget.load_video(self._project.video_path(self.video_list.selected_video))
```

Comment:
What happens if `self._project.video_path(self.video_list.selected_video)` returns `None`?

Reply:
if a project is opened successfully, there is always a selected video 
the clear cache menu is disabled unless a project has been loaded, so this shouldn't happen, but I'll add some more defensive programming